src/bo4e_generator/sqlparser.py

```diff
@@ -0,0 +1,235 @@
+"""
+Contains code to generate SQLModel classes from json schemas.
+Since the used tool doesn't support all features we need, we monkey patch some functions.
+"""
+import json
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, DefaultDict, Union
+
+import black
+import isort
+from jinja2 import Environment, FileSystemLoader
+
+from bo4e_generator.schema import SchemaMetadata
+
+
+def adapt_parse_for_sql(
+    input_directory: Path, namespace: dict[str, SchemaMetadata]
+) -> tuple[dict[str, SchemaMetadata], dict[str, Any], Path, dict[str, str]]:
+    """
+    Scans fields of parsed classes to modify them to meet the SQLModel specifics and to introduce relationships.
+    Returns additional information, an input path with modified json schemas and arguments for the parser
+    """
```

Comment:
Die Funktion ist ein wenig lang. Vor allem weil die Funktion ja mehrere verschiedene Sachen tut.
Ist hier noch ok, aber versuch tendenziell lieber paar Funktionen mehr als zu wenig anzulegen - ich weiß, ist manchmal nicht ganz einfach :) Aber es vereinfacht anderen (und auch dir selbst in paar Wochen / Monaten) wesentlich, den Code besser zu verstehen.

Reply:
Hab ich in 41435edf27f4f93e41c4cf5b5fdc1ead9d792d56 nochmal gekürzt/aufgeteilt